bitnami/solr/templates/exporter-svc.yaml

```diff
@@ -0,0 +1,36 @@
+{{- if .Values.exporter.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: solr-exporter
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  name: "{{ include "solr.exporter-name" . }}"
+  {{- if or .Values.exporter.service.annotations .Values.commonAnnotations }}
+  annotations:
+    {{- if .Values.exporter.service.annotations }}
+    {{ include "common.tplvalues.render" ( dict "value" .Values.exporter.service.annotations "context" $) | nindent 4 }}
+    {{- end }}
+    {{- if .Values.commonAnnotations }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- end }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  type: {{ .Values.exporter.service.type }}
+  {{ if eq .Values.service.type "LoadBalancer" }}
+  loadBalancerSourceRanges: {{ .Values.service.loadBalancerSourceRanges }}
+  {{ end }}
+  {{- if (and (eq .Values.service.type "LoadBalancer") (not (empty .Values.service.loadBalancerIP))) }}
+  loadBalancerIP: {{ .Values.service.loadBalancerIP }}
+  {{- end }}
+  ports:
+    - name: tcp-client
+      port: {{ .Values.exporter.service.port }}
+      protocol: TCP
+      targetPort: {{ .Values.exporter.port }}
```

Comment:
I don't see the port section in exporter-deployment. Use the named port instead when you create it (similar to what you do in svc-headless)